### ISSUE TITLE
us-core-race and us-code ethnicity FHIR export broken when a Person's race attribute is "hispanic"

### ISSUE THREAD

**Issue by eedrummer** [synthetichealth/synthea#572]
It appears that if Person's race attribute is "hispanic", it outputs incorrect information. This is only an issue for FHIR STU3 and R4.

The root of the problem is that the export code is using `==`  and `!=` for string comparison in java, which compares by reference and not by contents. Oddly, the DSTU2 exporter correctly uses the `String.equals` function. 

The result of all of this is that when a person has a race attribute of "hispanic", then the FHIR output for the us-core-race code will be "2135-2", which is a valid OMB Ethnicity Category, but not a valid OMB Race Category.

Additionally, every person output to FHIR STU3 or R4 appears to have a us-core-ethnicity code of "2186-5" because the equality check on the race attribute fails on reference.

Fixing export of us-core-ethnicity is trivial. However, there are a couple of options for fixing export of us-core-race:

1. When a person has a race attribute of "hispanic", omit the us-core-race attribute. This appears to be the intent of the existing code
2. Populate the us-core-race attribute with a random value weighted by the distribution of races reported in the US Census with Hispanic ethnicity.

Is there a preferred solution to the issue?

**Comment by jawalonoski** [comment]
OMB Race Category is strangely limited set. The extension in R4 includes other codes as well. See http://hl7.org/fhir/us/core/ValueSet-omb-race-category.html#expansion

Anyway, the solution is probably to have the extension have an `ombCategory` of `UNK` and a `detailed` value of `2131-1` (Other).

```json
{
  "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
  "extension": [
    {
      "url": "ombCategory",
      "valueCoding": {
        "system": "http://terminology.hl7.org/CodeSystem/v3-NullFlavor",
        "code": "UNK",
        "display": "Unknown"
      }
    }, {
      "url": "detailed",
      "valueCoding": {
        "system": "urn:oid:2.16.840.1.113883.6.238",
        "code": "2131-1",
        "display": "Other"
      }
    }, {
      "url": "text",
      "valueString": "Other"
    }
  ]
}
```

**Comment by jawalonoski** [comment]
Or... since unknown isn't quite right:

```json
{
  "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
  "extension": [
    {
      "url": "detailed",
      "valueCoding": {
        "system": "urn:oid:2.16.840.1.113883.6.238",
        "code": "2131-1",
        "display": "Other"
      }
    }, {
      "url": "text",
      "valueString": "Other"
    }
  ]
}
```

**Comment by jawalonoski** [comment]
Fixed with #577
